firmware_DavidBoard/Core/Src/TemperatureCalc.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * TempCalc.c
+ *
+ *  Created on: Mar 13, 2022
+ *      Author: JacobS
+ */
+
+#include "TemperatureCalc.h"
+
+/***********************************************************************************************************************
+ * Definitions
+ **********************************************************************************************************************/
+
+/***********************************************************************************************************************
+ * Variables
+ **********************************************************************************************************************/
+
+// HAL handle
+/*extern I2C_HandleTypeDef hi2c2;
+static uint8_t rawAdcData[16];
+static bool adcDataNew;
+static I2CBusStatus_t currentBusStatus;*/
+
+
+//static double V1;
+//static double R1;
+//static double T_K_inverse;
+static double TCelsius;
+static int V1000int;
+static int R1000int;
+
+
+/***********************************************************************************************************************
+ * Prototypes
+ **********************************************************************************************************************/
+
+int Temperature_ADCtoCelsius(ThermistorADC_t *ThermADCValues, DataBuffer_t *DataBuffer);
+
+
+/***********************************************************************************************************************
+ * Code
+ **********************************************************************************************************************/
+
+<<<<<<< HEAD
+int Temperature_ADCtoCelsius(ThermistorADC_t *ThermADCValues, DataBuffer_t *DataBuffer) //change arguments - don't pass ThermADCValues
+=======
+int Temperature_ADCtoCelsius(ThermistorADC_t *ThermADCValues, DataBuffer_t *DataBuffer)
+>>>>>>> e7bf7227ccea139905cf390d9675ba7e16bbdd18
+{
+    /* //used for min/max/avg - not needed if these calcs are done downstream
+    double temperatureMin = 0;
+    double temperatureMax = 0;
+    double temperatureSum = 0;
+    double temperatureAvg = 0;
+    */
+    for (int i = 0; i <= 3; i++) 
+    {
+        if((ThermADCValues->thermistor[i]) <= 0 || (ThermADCValues->thermistor[i]) >= ADC_QUANTIZATION-1)
+        {
+            return RETROFRIGERATION_FAILED;
+        }
+        /*//Outdated - uses too much memory
+        V1 = (ADC_VALUE*BOARD_V_FS)/(ADC_QUANTIZATION-1);
+        R1 = V1 * PCB_DIVIDING_RESISTANCE / (BOARD_V_FS - V1);
+
+        T_K_inverse = STEINHART_HART_A + STEINHART_HART_B*log(R1)+STEINHART_HART_C*pow(log(R1), 3); //Steinhart-Hart equation of thermistor temp-resistance relation. Accurate but uses lots of resources
+        TCelsius = 1/T_K_inverse-273.15;
+        */
+        V1000int = ((ThermADCValues->thermistor[i])*ADC_V_FS)/(ADC_QUANTIZATION-1);
```

Comment:
Did this actually work ? These are all integers, so integer math would have been invoked and sliced all your decimals.